gammapy/data/io.py

```diff
@@ -141,3 +143,137 @@ def to_hdu(self, event_list, format="gadf"):
+
+        table_disk = Table.read(filename, format="fits", hdu=hdu)
+        table_disk_meta = table_disk.meta
+
+        format = table_disk_meta.get("HDUCLASS", "unknown")
+        version = table_disk_meta.get("HDUVERS", "unknown")
+
+        if (format == "unknown") or (format is None):
+            format = "GADF"
+            version = "0.3"
+            warnings.warn(
+                UserWarning(
+                    f"Could not infer fileformat from metadata in {filename}, assuming GADF."
+                )
+            )
+
+        if format == "GADF" or format == "OGIP":
+            if version == "0.2" or version == "0.3":
+                return self.from_gadf_table(table_disk)
+            else:
+                raise ValueError(f"Unknown {format} version :{version}")
+        else:
+            raise ValueError(f"Unknown fileformat :{format}")
+
+    @staticmethod
+    def from_gadf_table(table_gadf):
+        """Convert gadf observation table into internal data format.
+
+        Parameters
+        ----------
+        table_gadf : `~astropy.Table.table`
+            Table in gadf 0.2/0.3 format.
+            https://gamma-astro-data-formats.readthedocs.io/en/v0.2/data_storage/obs_index/index.html
+            https://gamma-astro-data-formats.readthedocs.io/en/v0.3/data_storage/obs_index/index.html
+
+        Returns
+        -------
+        ObservationTable : `~gammapy.data.ObservationTable`
+            ObservationTable in internal data format.
+
+            Mandatory Columns: OBS_ID
+
+            Defined Units and Types for Columns as follows:
+
+            OBS_ID	RA_PNT	DEC_PNT	ALT_PNT	AZ_PNT	TSTART	TSTOP
+                        deg	    deg	    deg	   deg
+            int64   float64	float64	float64 float64	 Time	 Time
+        """
+
+        names_gadf = table_gadf.colnames
+        meta_gadf = table_gadf.meta
+
+        required_names_gadf = [
+            "OBS_ID",
+        ]
+
+        missing_names = set(required_names_gadf).difference(names_gadf)
+        if len(missing_names) != 0:
+            raise RuntimeError(
+                f"Not all columns required to read from GADF were found in file. Missing: {missing_names}"
+            )
+
+        removed_names = []
+
+        try:
+            obs_id = table_gadf["OBS_ID"].astype("int")
+        except ValueError:
+            raise RuntimeError(
+                "Could not convert OBS_ID to int. Can not create table without OBS_ID."
+            )
+        new_table = Table({"OBS_ID": obs_id}, meta=meta_gadf)
+        new_table = table.unique(new_table, keys="OBS_ID")
+        removed_names.append("OBS_ID")
+
+        for colname in ["RA_PNT", "DEC_PNT", "ALT_PNT", "AZ_PNT"]:
+            if colname in names_gadf:
+                try:
+                    new_table[colname] = Quantity(
+                        table_gadf[colname].astype("float64"),
+                        u.Unit(table_gadf[colname].unit),
+                    ).to(u.deg)
```

Comment:
```suggestion
                    new_table[colname] = table_gadf[colname].quantity.to("deg")
```
you don't really need to adapt the type